node_modules/oae-core/topnavigation/topnavigation.html

```diff
@@ -42,102 +42,31 @@ <h1 id="topnavigation-target" class="sr-only">__MSG__SKIP_MENU_TARGET__</h1>
 -->
 
 <div id="topnavigation-left-anonymous-template"><!--
-    {if oae.api.config.getValue('oae-authentication', 'local', 'allowAccountCreation')}
-        <button type="button" class="btn btn-link oae-trigger-register">__MSG__SIGN_UP__</button>
+    {if !hidesignup}
+        <a id="topnavigation-left-signup" href="/signup" class="btn btn-link">__MSG__SIGN_UP__</button>
     {/if}
 --></div>
 
 <div id="topnavigation-right-anonymous-template"><!--
-    {var hasLocalAuth = enabledStrategies['local'] || enabledStrategies['ldap']}
-    {var hasExternalAuth = enabledStrategies['cas'] || enabledStrategies['facebook'] || enabledStrategies['google'] ||
-                           enabledStrategies['googleApps'] || enabledStrategies['shibboleth'] || enabledStrategies['twitter']}
-
-    {var hasSingleInstitutionalAuth = _.keys(enabledStrategies).length === 1 && _.contains(['cas', 'googleApps', 'shibboleth'], _.keys(enabledStrategies)[0])}
-
-    {macro externalAuth(id, strategy, title, icon)}
-        <li>
-            <form action="${strategy.url}" method="POST">
-                <input type="hidden" name="redirectUrl" value="${redirectUrl|encodeForHTMLAttribute}" />
-                <button type="submit" class="btn btn-external-auth btn-${id}">
-                    {if icon}
-                        <i class="icon fa fa-${icon}"></i>
-                    {/if}
-                    {if strategy.name}
-                        {var strategyName = strategy.name}
-                        __MSG__SIGN_IN_WITH_STRATEGY__
-                    {else}
-                        ${title}
-                    {/if}
-                </button>
-            </form>
-        </li>
-    {/macro}
-
     <ul class="list-inline">
-        <li class="topnavigation-search-button">
-            <a href="/search" class="btn btn-link" title="__MSG__SEARCH__" role="menuitem">
-                <i class="fa fa-search"><span class="sr-only">__MSG__SEARCH__</span></i>
+        {if !hidesignup}
+            <li class="topnavigation-search-button">
+                <a href="/search" class="btn btn-link" title="__MSG__SEARCH__" role="menuitem">
```

Comment:
My previous comment was incorrect, I've confused it with the topnavigation enhancements, which is the PR that I rebased onto inviting guests. 

At the moment in `inviting-guests`, `oae.signup.css` has a style rule to hide the search form unconditionally. However in the topnav enhancements PR, that has all changed to be managed in `topnavigation.css`.